synapseclient/models/folder.py

```diff
@@ -427,34 +424,30 @@ async def copy_async(
 
         loop = asyncio.get_event_loop()
 
-        current_context = context.get_current()
         syn = Synapse.get_client(synapse_client=synapse_client)
         source_and_destination = await loop.run_in_executor(
             None,
-            lambda: run_and_attach_otel_context(
-                lambda: copy(
-                    syn=syn,
-                    entity=self.id,
-                    destinationId=parent_id,
-                    excludeTypes=exclude_types or [],
-                    skipCopyAnnotations=not copy_annotations,
-                    updateExisting=file_update_existing,
-                    setProvenance=file_copy_activity,
-                ),
-                current_context,
+            lambda: copy(
+                syn=syn,
+                entity=self.id,
+                destinationId=parent_id,
+                excludeTypes=exclude_types or [],
+                skipCopyAnnotations=not copy_annotations,
+                updateExisting=file_update_existing,
+                setProvenance=file_copy_activity,
             ),
         )
 
         new_folder_id = source_and_destination.get(self.id, None)
         if not new_folder_id:
             raise SynapseError("Failed to copy folder.")
         folder_copy = await (
-            await Folder(id=new_folder_id).get_async()
+            await Folder(id=new_folder_id).get_async(synapse_client=syn)
         ).sync_from_synapse_async(
             download_file=False,
             synapse_client=synapse_client,
```

Comment:
Shouldn't this be `synapse_client=syn` like above?

Reply:
They end up working the exact same way because of how `syn = Synapse.get_client(synapse_client=synapse_client)` is used. However, I made a change here to make this consistent